### PR TITLE
Add shared components documentation link

### PR DESCRIPTION
**Pull request by da1910**
This PR adds a link to the Shared Components documentation.

There we have documented the OpenAPI-Common Library, and intend to also document any other shared components that the wider PyAnsys team generates.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -544,6 +544,15 @@ For more details, see:
   - `Legacy PyMAPDL Reader GitHub <https://github.com/pyansys/pymapdl-reader>`_
 
 
+Shared Components
+-----------------
+The PyAnsys project publishes and consumes shared components which enable
```
Comment (Andy-Grigg) [inline README.rst:549]:
```suggestion
The PyAnsys project publishes and consumes shared software components. These enable
```

File: README.rst [hunk 2]
```diff
@@ -544,6 +544,15 @@ For more details, see:
...
+interoperability and minimise maintenance burden for our other packages.
```
Comment (Andy-Grigg) [inline README.rst:550]:
```suggestion
interoperability between PyAnsys packages and minimizes maintenance burden.
```

**Review by Andy-Grigg — CHANGES_REQUESTED** [review]
One minor wording change, just to make the sentence slightly simpler.

File: doc/source/index.rst [hunk 3]
```diff
@@ -7,6 +7,7 @@
    DPF-Core <https://dpfdocs.pyansys.com/>
    DPF-Post <https://postdocs.pyansys.com/>
    Legacy PyMAPDL Reader <https://readerdocs.pyansys.com/>
```
Comment (akaszynski) [inline doc/source/index.rst:9]:
```suggestion
   PyDPF-Core <https://dpfdocs.pyansys.com/>
   PyDPF-Post <https://postdocs.pyansys.com/>
   Legacy PyMAPDL Reader <https://readerdocs.pyansys.com/>
```

**Review by akaszynski — APPROVED** [review]
LGTM.

We're going to have to (re)consider how to organize this once the other projects come online.

**Pull request merged**